_includes/manuals/nightly/3.7_migration.md

```diff
@@ -0,0 +1,185 @@
+
+Full backup is absolute necessary before starting the process. We recommend
+both database dump and snapshot of whole operating system and data if possible
+when running in a VM or on a LVM.
+
+#### Prerequisites
+
+It's a good idea to delete unwanted records from the database. The smaller the
+database is, the faster will be the migration process. Typical offenders are
+audit records, facts and reports. Foreman ships with rake tasks and cron job to
+execute these cleanup tasks, but they can be executed manually. Steps are shown
+down below after all instances are stopped.
+
+Foreman database must be cleaned up from any leftovers from uninstalled
+plugins. When a Foreman plugin is activated, database migration is perfored and
+new tables and indices are created. However when a plugin is uninstalled and
+deactivated, no tables are ever dropped and the data is left in the database
+forever. It's good idea to review at least OS installed package log to find out
+if any foreman package was ever uninstalled. Unfortunately, there is no list of
+tables for each plugin.
+
+#### Preparing the environment
+
+This blog post assumes that you already have the source and target databases
+set up. Make sure to setup the Postgresql database with UTF-8 enabled.
+
+The first step is to install required dependencies:
+
+    yum install -y foreman-postgresql
+
+Next, stop all your foreman instances and make sure, they stay stopped for
+good. On Red Hat compatible systems, you need to stop the `httpd` and (if
+applicable) `foreman-tasks` service. There is a helper utility called
+`foreman-maintain` when installed to ease the process of shutting down
+services. Make sure that neither Puppet nor a monitoring system automatically
+starts these services again.
+
+You can issue the MySQL command `SHOW FULL PROCESSLIST;` to show you all active
+connections to your MySQL database. Make sure, there are none.
+
+TODO: (WHY DO WE NEED THIS): If you are using the foreman-tasks plugin, you
+need to patch the script we are using to migrate the data. If you are not sure,
+wether you use this plugin, check if you have an entry `Tasks` in Foreman's
+Monitor menu. If you are still not sure, it is safe to apply the patch in any
+case.
+
+```sh
+wget -O /tmp/convert-tasks-patch.diff https://gist.githubusercontent.com/timogoebel/cd29e5206678ca0e0d78222af70583ec/raw/bbe998624098c1c3e656fe2582d33b0d9f9285aa/0001-convert-skip-foreman-tasks-tables.patch
+yum install patch -y
+cd /usr/share/foreman
+
+patch -p1 < /tmp/convert-tasks-patch.diff
+```
+
+The foreman user that has access to your Postgresql database and is used for
+the conversion temporarily needs super user privileges.
+
+```bash
+## postgres@postgres:~  $ su - postgres
+## postgres@postgres:~  $ psql
+psql (9.6.2)
+Type "help" for help.
+
+postgres=# ALTER USER foreman WITH SUPERUSER;
+ALTER ROLE
+postgres=#
+```
+
+#### Migrating
+
+Before proceeding migration, it's a good idea to permanently delete all audit
```

Comment:
tbh I don't think we would like to ask users to delete their audits, only if they dont use it, I would just say it can speed things up, but recommend users to delete it.

Reply:
We do this via cron anyway both for reports and (now) for audits. I will update the heading.

Reply:
Marked the title as "optional" tho.